src/ledger-app.js

```diff
@@ -307,4 +309,76 @@ LedgerApp.prototype.showAddress = function (hrp, path) {
         process_error_response);
 };
 
+LedgerApp.prototype.getAddress = function (hrp, path) {
+    let data = Buffer.concat([serialize_hrp(hrp), serialize_path(path)]);
+    let buffer = serialize(CLA, INS_GET_ADDR_SECP256K1, 0, 0, data);
+
+    return this.comm.exchange(buffer.toString('hex'), [0x9000, 0x6A80]).then(
+        function (apduResponse) {
+            let result = {};
+            apduResponse = Buffer.from(apduResponse, 'hex');
+            let error_code_data = apduResponse.slice(-2);
+
+            result["return_code"] = error_code_data[0] * 256 + error_code_data[1];
+            result["error_message"] = errorMessage(result["return_code"]);
+
+            if (result.return_code !== 0x9000) {
+                return
+            }
+
+            result["compressed_pk"] = Buffer.from(apduResponse.slice(0, 33));
+            result["bech32_address"] = Buffer.from(apduResponse.slice(33, -2)).toString();
+
+            return result;
+        },
+        process_error_response);
+};
+
+LedgerApp.prototype.appInfo = function () {
+    let buffer = serialize(0xB0, 0x01, 0, 0);
+
+    return this.comm.exchange(buffer.toString('hex'), [0x9000, 0x6A80]).then(
+        function (apduResponse) {
+            let result = {};
+            apduResponse = Buffer.from(apduResponse, 'hex');
+            let error_code_data = apduResponse.slice(-2);
+
+            result["return_code"] = error_code_data[0] * 256 + error_code_data[1];
+            result["error_message"] = errorMessage(result["return_code"]);
+
+            if (result.return_code !== 0x9000) {
+                return
+            }
+
+            if (apduResponse[0] !== 1) {
+                result["error_message"] = "Invalid response - format ID not recognized";
+                return result
+            }
+
+            var idx = 1;
+            const appNameLen = apduResponse[idx];
```

Comment:
this should always be `1`:
```suggestion
            const appNameLen = apduResponse[1];
```